dvc/repo/ls.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+from dvc.exceptions import DvcException, OutputNotFoundError
+
+
+@staticmethod
+def ls(url, target=None, rev=None, recursive=None, outputs_only=False):
+    from dvc.external_repo import external_repo
+    from dvc.repo import Repo
+
+    with external_repo(url, rev) as repo:
+        result = set()
+        if isinstance(repo, Repo):
+            result.update(_ls_outs_repo(repo, target, recursive))
+        if not outputs_only:
+            try:
+                result.update(_ls_files_repo(repo, target, recursive))
+            except DvcException as exc:
+                if not len(result):
+                    raise exc
+        result = sorted(list(result))
+    return result
+
+
+def _ls_files_repo(repo, target=None, recursive=None):
+    from dvc.scm.tree import WorkingTree
+    from dvc.ignore import CleanTree
+
+    def prettify(prefix, filename):
+        if not prefix == os.curdir:
+            filename = os.path.join(prefix, filename)
+        return filename
+
+    files = []
+    target_path = (
+        os.path.join(repo.root_dir, target) if target else repo.root_dir
+    )
+    tree = CleanTree(WorkingTree(target_path))
+
+    try:
+        if not os.path.exists(target_path):
+            raise FileNotFoundError("{} doesn't exist".format(target_path))
```

Comment:
This error message is meaningless, you are catching it right away down below and re-raising it. You could raise what you want right away.